packages/oauth/src/form/form-service.ts

```diff
@@ -201,6 +233,16 @@ export class FormService implements OAuthProtoService {
         else log.error(err)
       }
 
+      //
+      // If token failed to be decoded then try and see if
+      // it is an openshift cluster and ask the API for the username
+      //
+      if (subject === this.userProfile.getToken()) {
+        // Will return either the username or null (if api request not valid)
+        // If null, subject will remain as the token
```

Comment:
It's still wrong to use the token as the fallback for a user name. "subject" should mean the subject retrieved from a JWT token, not a general token itself. We should rewrite the entire logic and use it only for the subject from JWT, and just use a default name for the user such as `user`.

Reply:
Why using token as user name is wrong is that it may expose the secret credential to the UI. It's like showing password to the UI. So taking screenshots for the UI may leak it. Users shouldn't need to take care about it.